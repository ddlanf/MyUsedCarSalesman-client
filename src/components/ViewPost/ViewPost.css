
.view-post-box{
  margin-top: 15%;
  margin-bottom: 10%;
  padding-top: 2%;
  width: 90%;
  margin-left: auto;
  margin-right: auto;
  border: 1px solid black;
  background-color: #DFE32E;
  border-radius: 5%;
}

.view-post-image{
  text-align: center;
  margin-top: 2%;
  margin-left: auto;
  margin-right: auto;
  width: 70%;
  display: block;
}

.view-post-image{
  margin-left: auto;
  margin-right: auto;
  width: 80%;
  max-width: 800px;
  border: #999090 1px solid;
}

.view-post-carInfo-name, .view-post-carInfo-price, .view-post-carInfo-commission{
  display: block;
  text-align: center;
  margin-left: auto;
  margin-right: auto;
  width: 70%;
  font-family: 'Merriweather Sans', sans-serif;
}

.view-post-carInfo-name{
  font-size: 7vw;
  margin-top: 3%;
  margin-bottom: 0;
}

.view-post-carInfo-price{
  font-size: 8vw;
  margin-top: 2%;
  margin-bottom: 1%;
}

.view-post-carInfo-commission{
  font-size: 5vw;
}

.view-post-info{
  list-style: none;
  font-size:  4vw;
  padding-left: 10%;
}

.view-post-text-box-label{
  font-weight: 600;
  font-size: 4.5vw;
}

.view-post-text-box{
  border: #999090 1px solid;
  background-color: #EAEDEC;
  width: 85%;
  font-size: 4.5vw;
  padding-left: 1%;
  margin-bottom: 2%;
}

.view-post-text-box.description, .view-post-text-box.terms-cond{
  max-height: 300px;
  height: 35vw;
  overflow-y: auto;
}

.view-post-contact-button{
  display: block;
  margin-left: auto;
  margin-right: auto;
  margin-bottom: 5%;
  font-size: 3vw;
  width: 30%;
  height: 10vw;
  border-radius: 10%;
  border: #999090 1px solid;
  background-color: #EAEDEC;
}

.view-post-contact-button:hover{
  background-color: #999090;
  transition: 1s;
}

.view-post-contact-button:visited{
  background-color: #999090;
}

.view-post-user-email{
  display: block;
  margin-left: auto;
  margin-right: auto;
  margin-top: 0;
  margin-bottom: 2%;
  text-align: center;
  font-size: 4vw;
  width: 50%;
  border: #999090 1px solid;
  background-color: #EAEDEC;
  overflow-x: auto;
}

@media screen and (min-width: 1000px) {

  .view-post-box{
    margin-top: 10vw;
    width: 90%;
    display: flex;
    flex-direction: row;
    align-items: flex-start;
    justify-content: space-evenly;
    flex-wrap: wrap;
    margin-left: auto;
    margin-right: auto;
  }

  .view-post-text-box{
    width: 50%;
    font-size: 1.5vw;
    padding-left: 1%;
    margin-bottom: 2%;
  }

  .view-post-text-box-label{
    font-size: 1.5vw;
  }

  .view-post-carInfo-name{
    font-size: 3vw;
  } 

  .view-post-carInfo-price{
    text-align: center;
    font-size: 2.7vw;
    width: 80%;  
  }

  .view-post-carInfo-commission{
    font-size: 25px;
  }

  .view-post-info{
    list-style: none;
    font-size:  0;
    padding-left: 0;
    width: 100%;
    display: flex;
    align-items: flex-start;
    justify-content: space-evenly;
  }

  .view-post-image{
    text-align: center;
    margin-top: 2%;
    margin-left: auto;
    margin-right: auto;
    width: 70%;
    display: block;
  }
  
  .view-post-image{
    width: 85%;
    max-width: 800px;
    border: #999090 1px solid;
  }

  .view-post-desk-top-view-box{
    width: 45%;
  }

  .view-post-desk-top-view-box-two{
    width: 55%;
    margin-top: 2%;
  }

  .view-post-desk-top-view-list-box{
    width: 50%;
  }

  .view-post-desk-top-view-list-box-two{
    width: 50%;
  }
  
  .view-post-text-box{
    width: 90%;
  }

  .view-post-contact-button{
    font-size: 20px;
    height: 50px;
    margin-top: 1%;
    margin-bottom: 2%;
    width: 300px;
    border-radius: 10%;
    border: #999090 1px solid;
    background-color: #EAEDEC;
  }

  .view-post-user-email{
    display: block;
    margin-left: 25%;
    margin-right: 25%;
    margin-top: 0;
    margin-bottom: 2%;
    text-align: center;
    font-size: 30px;
    width: 40%;
    border: #999090 3px solid;
    background-color: #EAEDEC;
    overflow-x: auto;
  }
}
