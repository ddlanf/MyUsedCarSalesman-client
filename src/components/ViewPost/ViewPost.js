import React, { Component } from 'react'
import { withRouter } from 'react-router-dom'
import PostApiService from '../../services/posts-api-service'
import ImageApiService from '../../services/images-api-service'
import UserApiService from '../../services/users-api-service'
import PostContext from '../../contexts/PostContext'
import TokenService from '../../services/token-service'
import './ViewPost.css'

class ViewPost extends Component {

    static contextType = PostContext
    constructor(props){
        super(props)
        this.state = { clicked : false, email: ''}
    }
   
    
    buttonClicked = () =>{
        this.setState({clicked : true})
        if(TokenService.hasAuthToken()){
            UserApiService.getUserEmail(this.context.post.user_name)
                .then(email => this.context.setEmail(email))
                .then(this.context.setEmail(''))
                .catch(
                    this.context.setError 
                )
        }
        else{ this.props.history.push('/login') }
    }

    componentDidMount(){
        const { postId } = this.props.match.params;
        this.context.clearError()
        PostApiService.getPost(postId)
            .then(post =>{ 
                this.context.setPost(post)
            })
            .catch(this.context.setError)

        ImageApiService.getImage(postId)
            .then(images => this.context.setImages(images))
            .catch(this.context.setError)
    }


    render() {

        const imageDefault = [{ "src": "", "alt":"" }]
        
        const  image =  this.context.images.length ? this.context.images : imageDefault
        const  carInfo = this.context.post
       
        return (
            <>
                <section className="view-post-box">
                    <div className="view-post-desk-top-view-box">
                        <h1 className="view-post-carInfo-name"> {carInfo.year} {carInfo.make} {carInfo.model}</h1>
                        <h2 className="view-post-carInfo-price">${carInfo.price}</h2>
                        <img
                                className="view-post-image"
                                src={image[0].src ? image[0].src : require('../../Utils/myusecarsalesman_pics/notfound.png') }
                                alt={image[0].alt ? image[0].alt : 'Not Found' }
                        />
                        <h3 className="view-post-carInfo-commission">{carInfo.commission_amount}</h3>
                    </div>
                    <div className="view-post-desk-top-view-box-two">
                        <ul className="view-post-info">
                            <div className="view-post-desk-top-view-list-box">
                                <li><span className="view-post-text-box-label">Make</span></li>
                                <li>
                                    <div className="view-post-text-box">
                                        {carInfo.make}
                                    </div>
                                </li>
                                <li><span className="view-post-text-box-label">Model</span></li>
                                <li>
                                    <div className="view-post-text-box">
                                        {carInfo.model}
                                    </div>
                                </li>            
                                <li><span className="view-post-text-box-label">Year</span></li>
                                <li>
                                    <div className="view-post-text-box">
                                        {carInfo.year}
                                    </div>
                                </li>            
                                <li><span className="view-post-text-box-label">Mileage</span></li>
                                <li>
                                    <div className="view-post-text-box">
                                        {carInfo.mileage}
                                    </div>
                                </li>            
                                <li><span className="view-post-text-box-label">Description</span></li>
                                <li>
                                    <div className="view-post-text-box description">
                                        {carInfo.description}
                                    </div>
                                </li>         
                            </div>   
                            <div className="view-post-desk-top-view-list-box-two">
                                <li><span className="view-post-text-box-label">Price</span></li>
                                <li>
                                    <div className="view-post-text-box">
                                        ${carInfo.price}
                                    </div>
                                </li>
                                <li><span className="view-post-text-box-label">Commission</span></li>
                                <li>
                                    <div className="view-post-text-box">
                                        {carInfo.commission_amount}
                                    </div>
                                </li>
                                <li><span className="view-post-text-box-label">Location</span></li>
                                <li>
                                    <div className="view-post-text-box">
                                        {carInfo.location}
                                    </div>
                                </li>
                                <li><span className="view-post-text-box-label">Other terms and conditions</span></li>
                                <li>
                                    <div className="view-post-text-box terms-cond">
                                        {carInfo.other_terms_and_conditions}
                                    </div>
                                </li>
                                <li><span className="view-post-text-box-label">Posted By</span></li>
                                <li>
                                    <div className="view-post-text-box">
                                        {carInfo.user_name}  
                                    </div>
                                </li>
                            </div>
                        </ul>
                    </div>
                    <button 
                        className="view-post-contact-button"
                        onClick={this.buttonClicked}>
                            Contact Owner
                    </button>
                    {this.state.clicked ? <div className="view-post-user-email">{this.context.email}</div> : ''}
                </section>
            </>
        )
    }
}


export default withRouter(ViewPost)