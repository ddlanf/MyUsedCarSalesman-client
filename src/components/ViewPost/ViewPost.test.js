import React from 'react';
import ReactDOM from 'react-dom';
import ViewPost from './ViewPost';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom'

describe('ViewPost component', () => {
    
    it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<BrowserRouter>
                        <ViewPost/>
                    </BrowserRouter>, div);
    ReactDOM.unmountComponentAtNode(div);
    });

    it('renders the UI as expected', () => {
        const tree = renderer
          .create(
                <BrowserRouter>
                    <ViewPost/>
                </BrowserRouter>)
          .toJSON();
        expect(tree).toMatchSnapshot();  
      });
});