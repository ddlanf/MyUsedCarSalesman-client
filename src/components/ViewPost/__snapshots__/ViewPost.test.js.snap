// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ViewPost component renders the UI as expected 1`] = `
<section
  className="view-post-box"
  style={
    Object {
      "opacity": "0",
    }
  }
>
  <div
    className="view-post-desk-top-view-box"
  >
    <h1
      className="view-post-carInfo-name"
    >
       
       
       
    </h1>
    <h2
      className="view-post-carInfo-price"
    >
      $
    </h2>
    <img
      alt="Not Found"
      className="view-post-image"
      src="notfound.png"
    />
    <h3
      className="view-post-carInfo-commission"
    />
  </div>
  <div
    className="view-post-desk-top-view-box-two"
  >
    <ul
      className="view-post-info"
    >
      <div
        className="view-post-desk-top-view-list-box"
      >
        <li>
          <span
            className="view-post-text-box-label"
          >
            Make
          </span>
        </li>
        <li>
          <div
            className="view-post-text-box"
          />
        </li>
        <li>
          <span
            className="view-post-text-box-label"
          >
            Model
          </span>
        </li>
        <li>
          <div
            className="view-post-text-box"
          />
        </li>
        <li>
          <span
            className="view-post-text-box-label"
          >
            Year
          </span>
        </li>
        <li>
          <div
            className="view-post-text-box"
          />
        </li>
        <li>
          <span
            className="view-post-text-box-label"
          >
            Mileage
          </span>
        </li>
        <li>
          <div
            className="view-post-text-box"
          />
        </li>
        <li>
          <span
            className="view-post-text-box-label"
          >
            Description
          </span>
        </li>
        <li>
          <div
            className="view-post-text-box description"
          />
        </li>
      </div>
      <div
        className="view-post-desk-top-view-list-box-two"
      >
        <li>
          <span
            className="view-post-text-box-label"
          >
            Price
          </span>
        </li>
        <li>
          <div
            className="view-post-text-box"
          >
            $
          </div>
        </li>
        <li>
          <span
            className="view-post-text-box-label"
          >
            Commission
          </span>
        </li>
        <li>
          <div
            className="view-post-text-box"
          />
        </li>
        <li>
          <span
            className="view-post-text-box-label"
          >
            Location
          </span>
        </li>
        <li>
          <div
            className="view-post-text-box"
          />
        </li>
        <li>
          <span
            className="view-post-text-box-label"
          >
            Other terms and conditions
          </span>
        </li>
        <li>
          <div
            className="view-post-text-box terms-cond"
          />
        </li>
        <li>
          <span
            className="view-post-text-box-label"
          >
            Posted By
          </span>
        </li>
        <li>
          <div
            className="view-post-text-box"
          />
        </li>
      </div>
    </ul>
  </div>
  <button
    className="view-post-contact-button"
    onClick={[Function]}
  >
    Contact Owner
  </button>
  
</section>
`;
